### PR TITLE
Add Image Transformer doc and some fix

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

Add Image Transformer doc and some fix

## How was this patch tested?

unit test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: docs/docs/APIGuide/Data.md [hunk 1]
```diff
@@ -391,3 +391,153 @@ Output may be
...
 ```
+
+---
+## **ImageFeature**
+`ImageFeature` is a representation of one image.
+It can include various status of an image, by using key-value store.
+The key is string that identifies the corresponding value.
+Some predefined keys are listed as follows:
+* uri: uri that identifies image
+* mat: image in OpenCVMat
```
Comment (jason-dai) [inline docs/docs/APIGuide/Data.md]: Do you have API doc for OpenCVMat?

File: docs/docs/APIGuide/Data.md [hunk 2]
```diff
@@ -391,3 +391,153 @@ Output may be
...
+* bytes: image file in bytes
+* floats: image pixels in float array
+* size: current image size (height, width, channel)
+* originalSize: original image size (height, width, channel)
+* label: image label
+* predict: image prediction result
+* boundingBox: store boundingBox of current image,
+it may be used in crop/expand that may change the size of image
+* sample: sample
+* imageTensor: Image Tensor
```
Comment (jason-dai) [inline docs/docs/APIGuide/Data.md]: image pixels in Tensor?

File: docs/docs/APIGuide/Transformer.md [hunk 3]
```diff
@@ -13,3 +13,809 @@ trait Transformer[A, B] extends Serializable {
...
+**Python example:**
+```python
+from bigdl.transform.vision.image import *
+data = ImageFrame.read("/tmp/test.jpg")
+transformer = MatToFloats()
+transformed = transformer(data)
+```
+
+---
+## **MatToTensor**
```
Comment (jason-dai) [inline docs/docs/APIGuide/Transformer.md]: do we release mat here?

File: docs/docs/APIGuide/Data.md [hunk 4]
```diff
@@ -391,3 +391,153 @@ Output may be
...
+* mat: image in OpenCVMat
+* bytes: image file in bytes
+* floats: image pixels in float array
+* size: current image size (height, width, channel)
+* originalSize: original image size (height, width, channel)
+* label: image label
+* predict: image prediction result
+* boundingBox: store boundingBox of current image,
+it may be used in crop/expand that may change the size of image
+* sample: sample
```
Comment (jason-dai) [inline docs/docs/APIGuide/Data.md]: image (and label if available) stored as Sample?

File: docs/docs/APIGuide/Transformer.md [hunk 5]
```diff
@@ -13,3 +13,809 @@ trait Transformer[A, B] extends Serializable {
...
+val transformer = ImageFrameToSample(inputKeys: Array[String] = Array(ImageFeature.imageTensor),
+                               targetKeys: Array[String] = null,
+                               sampleKey: String = ImageFeature.sample)
+```
+**Python:**
+```python
+transformer = ImageFrameToSample(input_keys=["imageTensor"], target_keys=None,
+                                           sample_key="sample")
+```
+Transform imageframe to samples
```
Comment (jason-dai) [inline docs/docs/APIGuide/Transformer.md]: Is this correct?

**Comment by SeaOfOcean** [comment]
jenkins validation pass

**Pull request merged**